README.md

```diff
@@ -14,134 +20,432 @@ For suggestions on improvements,
-Please avoid using the dark theme if possible.
-
-### Command-Line Interface (CLI)
-
-First, enter the template root directory and set the profile via environment variable
+Model develompent consists of the following sequential steps:
 ```
-cd mlp-regression-template
+ingest -> split -> transform -> train -> evaluate -> register
 ```
+
+The batch scoring workflow consists of the following sequential steps:
 ```
-export MLFLOW_PIPELINES_PROFILE=local
+ingest -> predict
 ```
+A detailed reference for each step follows.
+### Step artifacts
+Each of the steps in the pipeline produces artifacts after completion. These artifacts consist of cards containing
+detailed execution information, as well as other step-specific information.
+The [`Pipeline.inspect()`](https://mlflow.org/docs/latest/python_api/mlflow.pipelines.html#mlflow.pipelines.regression.v1.pipeline.RegressionPipeline.inspect)
+API is used to view step cards. The [`get_artifact`](https://mlflow.org/docs/latest/python_api/mlflow.pipelines.html#mlflow.pipelines.regression.v1.pipeline.RegressionPipeline.get_artifact)
+API is used to load all other step artifacts by name.  
+Per-step artifacts are further detailed in the following step references.
 
-Then, try running the
-following [MLflow Pipelines CLI](https://mlflow.org/docs/latest/cli.html#mlflow-pipelines)
-commands to get started.
-Note that the `--step` argument is optional.
-Pipeline commands without a `--step` specified act on the entire pipeline instead.
+### Ingest step
+The ingest step resolves the dataset specified by the `data` section in [`pipeline.yaml`](https://github.com/mlflow/mlp-regression-template/blob/main/pipeline.yaml)
+and converts it to parquet format, leveraging the custom loader code specified in the `data` section if necessary.  
+**Note**: If you make changes to the dataset referenced by the ingest step (e.g. by adding new records or columns), 
+you must manually re-run the ingest step in order to use the updated dataset in the pipeline. 
+The ingest step does not automatically detect changes in the dataset.
 
-Available step names are: `ingest`, `split`, `transform`, `train`, `evaluate` and `register`.
+The custom loader function allows use of datasets in other formats, such as `csv`. 
+The function should be defined in [`steps/ingest.py`](https://github.com/mlflow/mlp-regression-template/blob/main/steps/ingest.py),
+and should accept two parameters:
+- `file_path`: `str`. Path to the dataset file.
+- `file_format`: `str`. The file format string, such as `"csv"`.
 
-- Display the help message:
-```
-mlflow pipelines --help
-```
+It should return a Pandas DataFrame representing the content of the specified file. [`steps/ingest.py`](https://github.com/mlflow/mlp-regression-template/blob/main/steps/ingest.py) contains an example placeholder function.
 
-- Run a pipeline step or the entire pipeline:
-```
-mlflow pipelines run --step step_name
-```
+#### Data
```

Comment:
```suggestion
#### Configuration
```

Can we make a `Configuration` subsection for each step section and then make all of its contents collapsible?

Reply:
Each step section already has a collapsed reference; it's just called "reference" instead of "configuration" right now

Reply:
Unless you mean we should move _all_ configuration-related content for each step (except for step overview) into the collapsible section? Right now only the yaml-related stuff is collapsed.

Reply:
I think we should change `Reference` to `Configuration` reference and make the text bold and bigger. It's hard to know what it contains right now and hard to see it / click it.

Reply:
Made it bold and underlined; don't think text sizing works without misaligning the text against the expander arrow.